src/test/java/com/squareup/javapoet/TypeNameTest.java

```diff
@@ -178,5 +179,6 @@ private void assertEqualsHashCodeAndToString(TypeName a, TypeName b) {
     assertEquals(a.toString(), b.toString());
     assertThat(a.equals(b)).isTrue();
     assertThat(a.hashCode()).isEqualTo(b.hashCode());
+    assertFalse(a.equals(null));
```

Comment:
True, but that's just an optimization. If `a.equals(b)` works then `b.equals(a)` must also work, thus we then know `a.equals(a)` will work.